File: Workshop/JavaBasicsProject/src/Basics/week_02/DataStructures/PersonalFinance.java
package Basics.week_02.DataStructures;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class PersonalFinance {
  public static void main(String[] args) {
    //    We are going to represent our expenses in a list containing integers.
    //
    //    Create a list with the following items.
    //    500, 1000, 1250, 175, 800 and 120
    //    Create an application which solves the following problems.

    List<Integer> expenses = new ArrayList<>(Arrays.asList(500, 1000, 1250, 175, 800, 120));

    //    How much did we spend?
    int total = 0;
    for (Integer spent : expenses) {
      total += spent;
    }
    System.out.println("Your total expenses are: " + total);
    //        Which was our greatest expense?
    int greatest = expenses.get(0);
    for (Integer max : expenses) {
      if (max > greatest) {
        greatest = max;
      }
    }
    System.out.println("Your greatest espense was: " + greatest);
    //    Which was our cheapest spending?
    int cheapest = expenses.get(0);
    for (Integer min : expenses) {
      if (min < cheapest) {
        cheapest = min;
      }
    }
    System.out.println("Your cheapest spending was: " + cheapest);

    //     What was the average amount of our spendings?
    int average = total / expenses.size();
    System.out.println("Your average spending was: " + average);

  }


}

File: Foundation/week_02-week_04/PythonBasicsProject/week_02/data_structures/personal_finance.py
# Personal finance
# We are going to represent our expenses in a list containing integers
# - Create a list with the following items.
#   - 500, 1000, 1250, 175, 800 and 120
expenses = [500, 1000, 1250, 175, 800, 120]


# - Create an application which solves the following problems.
#   - How much did we spend?
def total():
    sum = 0
    for expense in expenses:
        sum += expense
    return sum


print(total())


#   - Which was our greatest expense?
def greatest():
    greatest_expense = 0
    for expense in expenses:
        if expense > greatest_expense:
            greatest_expense = expense
    return greatest_expense


print(greatest())


#   - Which was our cheapest spending?
def cheapest():
    cheapest_expense = greatest()
    for expense in expenses:
        if expense < cheapest_expense:
            cheapest_expense = expense
    return cheapest_expense


print(cheapest())


#   - What was the average amount of our spendings?
def average():
    return total() / len(expenses)


print(average())
